datastream-utils/src/main/java/com/linkedin/datastream/common/zk/ZkClient.java

```diff
@@ -86,71 +80,20 @@ public ZkClient(String zkServers, int sessionTimeoutMs, int connectionTimeoutMs,
     _zkSessionTimeoutMs = sessionTimeoutMs;
   }
 
-  @Override
-  public void close() throws ZkInterruptedException {
-    if (LOG.isTraceEnabled()) {
-      StackTraceElement[] calls = Thread.currentThread().getStackTrace();
-      LOG.trace("closing zkclient. callStack: {}", Arrays.asList(calls));
-    }
-    getEventLock().lock();
-    try {
-      if (_connection == null) {
-        return;
-      }
-      LOG.info("closing zkclient: {}", ((ZkConnection) _connection).getZookeeper());
-      super.close();
-    } catch (ZkInterruptedException e) {
-      /*
-       * Workaround for HELIX-264: calling ZkClient#disconnect() in its own eventThread context will
-       * throw ZkInterruptedException and skip ZkConnection#disconnect()
-       */
-      try {
-        /*
-         * ZkInterruptedException#construct() honors InterruptedException by calling
-         * Thread.currentThread().interrupt(); clear it first, so we can safely disconnect the
-         * zk-connection
-         */
-        Thread.interrupted();
-        _connection.close();
-        /*
-         * restore interrupted status of current thread
-         */
-        Thread.currentThread().interrupt();
-      } catch (InterruptedException e1) {
-        throw new ZkInterruptedException(e1);
-      }
-    } finally {
-      getEventLock().unlock();
-      LOG.info("closed zkclient");
-    }
-  }
-
+  /**
+   * Check if a zk path exists. Changes the access modified to public, its defined as protected in parent class.
+   */
   @Override
   public boolean exists(final String path, final boolean watch) {
-    long startT = System.nanoTime();
-
-    try {
-      return retryUntilConnected(() -> _connection.exists(path, watch));
-    } finally {
-      long endT = System.nanoTime();
-      if (LOG.isTraceEnabled()) {
-        LOG.trace("exists, path: {}, time: {} ns", path, (endT - startT));
-      }
-    }
+    return super.exists(path, watch);
   }
 
+  /**
+   * Get all children of zk path. Changes the access modified to public, its defined as protected in parent class.
+   */
   @Override
   public List<String> getChildren(final String path, final boolean watch) {
-    long startT = System.nanoTime();
-
-    try {
-      return retryUntilConnected(() -> _connection.getChildren(path, watch));
-    } finally {
-      long endT = System.nanoTime();
-      if (LOG.isTraceEnabled()) {
-        LOG.trace("getChildren, path: {}, time: {} ns", path, (endT - startT));
-      }
-    }
+    return super.getChildren(path, watch);
   }
 
```

Comment:
Is the only reason for keeping this class around `ensureReadData`? I was just wondering if there's a substitute for that method in the base apache helix zookeeper client, because it seems like a generic/common operation.

Reply:
I couldn't find a equivalent implementation for that in Helix. So I kept those around.